internal/models/user.go

```diff
@@ -730,10 +730,19 @@ func (u *User) SoftDeleteUserIdentities(tx *storage.Connection) error {
 	return nil
 }
 
-func obfuscateFieldForSoftDelete(field string) string {
-	if field != "" {
-		softDeleteId, _ := crypto.GenerateNanoId(5)
-		return fmt.Sprintf("%s-%x", softDeleteId, sha256.Sum256([]byte(field)))
-	}
-	return field
+func obfuscateValue(id uuid.UUID, value string) string {
+	hash := sha256.Sum256([]byte(id.String() + value))
+	return base64.RawURLEncoding.EncodeToString(hash[:])
+}
+
+func obfuscateEmail(u *User, email string) string {
+	return obfuscateValue(u.ID, email)
+}
+
+func obfuscatePhone(u *User, phone string) string {
+	return obfuscateValue(u.ID, phone)[:15]
```

Comment:
Consider converting 15 to a constant or adding a comment explaining the rationale behind 15?
```suggestion
        // Field converted from VARCHAR(15) to text
	return obfuscateValue(u.ID, phone)[:15]
```

Worried that some developer in the future (maybe future version of us) might be briefly confused about the number